pkg/cluster/backupmanager/s3.go

```diff
@@ -2,34 +2,87 @@ package backupmanager
 
 import (
 	"fmt"
+	"io/ioutil"
+	"os"
 
+	"github.com/coreos/etcd-operator/pkg/backup/s3"
 	"github.com/coreos/etcd-operator/pkg/backup/s3/s3config"
 	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
 
 	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
 )
 
 type s3BackupManager struct {
 	s3config.S3Context
+	clusterName string
+	namespace   string
+	kubecli     *unversioned.Client
+	s3cli       *s3.S3
 }
 
-func NewS3BackupManager(s3Ctx s3config.S3Context) BackupManager {
-	return &s3BackupManager{
-		S3Context: s3Ctx,
+func NewS3BackupManager(s3Ctx s3config.S3Context, kubecli *unversioned.Client, clusterName, ns string) (BackupManager, error) {
+	cm, err := kubecli.ConfigMaps(ns).Get(s3Ctx.AWSConfig)
+	if err != nil {
+		return nil, err
 	}
+	se, err := kubecli.Secrets(ns).Get(s3Ctx.AWSSecret)
+	if err != nil {
+		return nil, err
+	}
+	homedir := os.Getenv("HOME")
+	if err := createDirIfNotExist(homedir + "/.aws"); err != nil {
```

Comment:
ok. can you do this in another func, like setup s3 or something? this makes the new func messy.